datacapture/src/main/java/com/google/android/fhir/datacapture/QuestionnaireViewModel.kt

```diff
@@ -198,23 +198,28 @@ internal class QuestionnaireViewModel(state: SavedStateHandle) : ViewModel() {
               )
             }
           if (enabled) {
-            listOf(
-              QuestionnaireItemViewItem(questionnaireItem, questionnaireResponseItem) {
-                questionnaireResponseItemChangedCallback(questionnaireItem.linkId)
-              }
-            ) +
-              getQuestionnaireState(
-                  questionnaireItemList = questionnaireItem.item,
-                  questionnaireResponseItemList =
-                    if (questionnaireResponseItem.answer.isEmpty()) {
-                      questionnaireResponseItem.item
-                    } else {
-                      questionnaireResponseItem.answer.first().item
-                    },
-                  // we're now dealing with nested items, so pagination is no longer a concern
-                  pagination = null,
-                )
-                .items
+            // TODO Issue#755 update question item with initial value, there is separate PR.
+            if (questionnaireItem.isHidden) {
+              emptyList()
+            } else {
+              listOf(
+                QuestionnaireItemViewItem(questionnaireItem, questionnaireResponseItem) {
+                  questionnaireResponseItemChangedCallback(questionnaireItem.linkId)
+                }
+              ) +
+                getQuestionnaireState(
+                    questionnaireItemList = questionnaireItem.item,
+                    questionnaireResponseItemList =
+                      if (questionnaireResponseItem.answer.isEmpty()) {
+                        questionnaireResponseItem.item
+                      } else {
+                        questionnaireResponseItem.answer.first().item
+                      },
+                    // we're now dealing with nested items, so pagination is no longer a concern
+                    pagination = null,
+                  )
+                  .items
+            }
           } else {
             emptyList()
           }
```

Comment:
Can be simplified to: 
```suggestion
            if (enabled && !questionnaireItem.isHidden) {
              listOf(
                QuestionnaireItemViewItem(questionnaireItem, questionnaireResponseItem) {
                  questionnaireResponseItemChangedCallback(questionnaireItem.linkId)
                }
              ) +
                getQuestionnaireState(
                    questionnaireItemList = questionnaireItem.item,
                    questionnaireResponseItemList =
                      if (questionnaireResponseItem.answer.isEmpty()) {
                        questionnaireResponseItem.item
                      } else {
                        questionnaireResponseItem.answer.first().item
                      },
                    // we're now dealing with nested items, so pagination is no longer a concern
                    pagination = null,
                  )
                  .items
            }
          } else {
            emptyList()
          }
```

Reply:
Actually, below edge case is also required.
If hidden extension is true and question item is enabled, then initial answers (if present) will be part of questionnaire response.
(Line no. 201 change is required if it is enabled and hidden.)